src/Init/Data/Prod.lean

```diff
@@ -7,5 +7,7 @@ prelude
 import Init.SimpLemmas
 
 instance [BEq α] [BEq β] [LawfulBEq α] [LawfulBEq β] : LawfulBEq (α × β) where
-  eq_of_beq {a b} h := by cases a; cases b; simp [BEq.beq] at h; have ⟨h₁, h₂⟩ := h; rw [eq_of_beq h₁, eq_of_beq h₂]
+  eq_of_beq {a b} (h : a.1 == b.1 && a.2 == b.2) := by
+    cases a; cases b
+    refine congr (congrArg _ (eq_of_beq ?_)) (eq_of_beq ?_) <;> simp_all
```

Comment:
Same here.

Reply:
Those are from the first commit, which added the `beq_iff_eq` simp lemma.  The previous proof used a "nonterminal simp", which are very likely to break when changing the library.